### ISSUE TITLE
BG matchmaker could use an update

### ISSUE THREAD

**Issue by Balanarr** [SunstriderEmu/BugTracker#2941]
As you may know (hopefully you do know), more than half of the bg's are a total joke to anyone who is not premading. For horde the queue pops in 1-3 mins and almost always starting with 2-4 horde vs 8-10+ alliance.

I propose the following updates or something similar if you have better ideas:
- Make the matchmaker wait until more slots are filled, like at least 8 players on each side for WSG, at least 12 on each side for AB,EOTS etc.
- Send equal amounts of bg pops for each side. Like if there are 20 alliance and 12 horde in queue for AB, send 12 for alliance and 12 for horde and wait and then as more horde queue, send 1 to alliance and 1 to horde etc.
While the server has a very very good population it's still not enough to assume that the spots will be filled and it is not, at least not until it is too late
- Prevent other bg pops while already in a bg or remove players from all other bg queues when entering a bg. While not blizzlike (at least not classic blizzlike), I think this will lead to more healthy bg's. Nothing sucks more than finally getting a full team and have 80% leave because av poped. Again for the same reason as above, there are just not enough players to fill the gaps.

**Comment by Jerrythemagicdog** [comment]
Thanks for the suggestions.

> Make the matchmaker wait until more slots are filled, like at least 8 players on each side for WSG, at least 12 on each side for AB,EOTS etc.

This one was already done since week two of the server but it doesn't help if people don't actually join the BG.
